swig/cpp/src/Tree_Data.cpp

```diff
@@ -520,35 +520,50 @@ S_Value Attr::value() {
 }
 S_Attr Attr::next() LY_NEW(attr, next, Attr);
 
-Difflist::Difflist(struct lyd_difflist *diff, S_Deleter deleter) {
-    diff = diff;
-    deleter = std::make_shared<Deleter>(diff, deleter);
+Difflist::Difflist(struct lyd_difflist *diff, S_Deleter deleter)
+  : diff(diff)
+  , deleter(std::make_shared<Deleter>(diff, deleter))
+{
 }
 Difflist::~Difflist() {};
 std::vector<S_Data_Node> Difflist::first() {
     std::vector<S_Data_Node> s_vector;
     unsigned int i = 0;
+    unsigned int s = 0;
+
+    for(s = 0; diff->type[s] != LYD_DIFF_END; s++) {
+    }
+
+    s_vector.reserve(s);
 
-    if (!*diff->first) {
+    if (diff->first == nullptr) {
+        s_vector.resize(s);
```

Comment:
This resizes an already empty vector to be empty again, and therefore it can be removed.

Reply:
Example uses:
```c++
void diff(std::shared_ptr<Difflist> ly_diff)
{
  auto type { ly_diff->type() };
  auto first { ly_diff->first() };
  auto second { ly_diff->second() };
  for (auto i { 0U }; type[i] != LYD_DIFF_END; ++ i)
  {
    std::cout << "i = " << i << " | " << type[i];
    {
      auto data { first[i] };
      if (data)
      {
        std::cout << ", first = " << data << '/' << data->schema()->name() << '/' << data->schema()->nodetype();
      }
    }
    {
      auto data { second[i] };
      if (data)
      {
        std::cout << ", second = " << data << '/' << data->schema()->name() << '/' << data->schema()->nodetype();
      }
    }
    std::cout << std::endl;
  }
  std::cout << "=======================" << std::endl;
}
```
Example result:
```i = 0 | 4, second = 0x18eb590/interfaces/16
i = 1 | 4, second = 0x18e9fb0/interfaces/16
i = 2 | 4, second = 0x18e71e0/interfaces/16
i = 3 | 4, second = 0x18e71a0/interfaces/16
i = 4 | 4, second = 0x18eb2b0/interfaces/16
i = 5 | 4, second = 0x18eb2f0/interfaces/16
i = 6 | 4, second = 0x18eb3a0/interfaces/16
i = 7 | 4, second = 0x18eb3e0/interfaces/16
i = 8 | 4, second = 0x18eb420/interfaces/16
i = 9 | 4, second = 0x18eb460/interfaces/16
i = 10 | 4, second = 0x190e620/interfaces/16
i = 11 | 4, second = 0x190e660/interfaces/16
i = 12 | 4, second = 0x190e6a0/interfaces/16
i = 13 | 4, second = 0x190e6e0/interfaces/16
i = 14 | 4, second = 0x190eed0/interfaces/16
i = 15 | 4, second = 0x190ef10/interfaces/16
i = 16 | 4, second = 0x190ef50/interfaces/16
i = 17 | 4, second = 0x190ef90/interfaces/16
i = 18 | 4, second = 0x190efd0/interfaces/16
i = 19 | 4, second = 0x190f010/interfaces/16
i = 20 | 4, second = 0x190f050/ports/16
i = 21 | 4, second = 0x190f090/ports/16
i = 22 | 4, second = 0x190f0d0/ports/16
i = 23 | 4, second = 0x190f110/ports/16
i = 24 | 4, second = 0x190f150/ports/16
i = 25 | 4, second = 0x190f190/ports/16
i = 26 | 4, second = 0x190f1d0/ports/16
i = 27 | 4, second = 0x190f210/ports/16
i = 28 | 4, second = 0x190f250/wire/16
i = 29 | 4, second = 0x190f290/wire/16
i = 30 | 4, first = 0x18ec120/mng-if/1, second = 0x190f2d0/ipv4/4
i = 31 | 4, first = 0x18ec410/mng-if/1, second = 0x190f310/gateway/4
```

Reply:
I do not understand this comment. I pointed out that the early return block is doing useless work -- resizing an empty vector to be empty again.